packages/lesmis/src/api/queries/useMapOverlayReportData.js

```diff
@@ -122,19 +148,21 @@ export const useMapOverlayReportData = ({ entityCode, year }) => {
   const { data: entitiesData, entitiesByCode, isLoading: entitiesLoading } = useEntitiesData(
     entityCode,
   );
+  const { data: overlaysData, isLoading: overlaysLoading } = useMapOverlaysData({ entityCode });
 
   const entityData = entitiesByCode[entityCode];
-
+  const overlay = findOverlay(overlaysData, selectedOverlay);
   const { startDate, endDate } = yearToApiDates(year);
 
   const params = {
     startDate,
     endDate,
     shouldShowAllParentCountryResults: false,
     type: 'mapOverlay',
+    legacy: overlay?.legacy,
```

Comment:
Because it's pulled from the `useMapOverlaysData` react hook which doesn't return a value right away

The query after this checks `!!overlay` before running so it's guaranteed to be set before it's actually used, but not to be set in this particular line where we assign

At least that's how my limited understanding of react goes, I do know that this line crashes if you don't account for it 😅

Reply:
Ah yep that makes sense. In that case I think we should avoid running any of the code below `useMapOverlaysData` while that's still loading. I'd jump to an early return after line 152 if `overlaysLoading` was true with just `return { isLoading: true };`, but that expects that the downstream consumers are ok with not having the other keys defined while it's loading, and I'm not sure whether that's true. Could you check it out?

Reply:
Can't do that because react needs to render the same number of hooks everytime. We can't exit before we've run all the `use` functions. That's why we have to pass it into the `enabled` param to check if we should actually perform the action.

Reply:
Ah yes, of course! Cool :+1: